src/store/reducers/subjects.js

```diff
@@ -44,7 +49,11 @@ const subjects = (state = { primary: [], lingual: [] }, action) => {
               ...subject,
               name: action.payload.name,
               elementaryScore: action.payload.elementaryScore,
-              advancedScore: action.payload.advancedScore
+              advancedScore: action.payload.advancedScore,
+              max: getSubjectScore({
```

Comment:
Computing max value should be done in action, so looking at payload we can see exactly which values are changing.